sds_data_manager/lambda_code/efs_lambda/lambda_function.py

```diff
@@ -0,0 +1,140 @@
+import os
+import shutil
+
+import boto3
+
+# Define the paths
+mount_path = os.environ.get("EFS_MOUNT_PATH")
+
+attitude_symlink_path = f"{mount_path}/latest_attitude_kernel.ah.a"
+ephemeris_symlink_path = f"{mount_path}/latest_ephemeris_kernel.bsp"
+
+
+def create_symlink(source_path: str, destination_path: str) -> None:
+    """Create a symlink from source_path to destination_path
+
+    Parameters
+    ----------
+    source_path : str
+        Source path of the symlink
+    destination_path : str
+        Destination path of the symlink
+    """
+
+    # Check if the symlink exists
+    if os.path.islink(destination_path):
+        # Remove the old symlink
+        os.unlink(destination_path)
+
+    # Create a new symlink pointing to the new file
+    os.symlink(source_path, destination_path)
```

Comment:
Use pathlib for all the path manipulations: https://docs.python.org/3/library/pathlib.html#pathlib.Path.symlink_to

Reply:
how does this know the source file that we want to link to the target symlink? Also curious of preference of Path over os. Can we use os and path in combination? Eg. create link using Path but read in its real path using os or do you need to do everything by one way?

Reply:
I think it would be this: `destination_path.symlink_to(source_path)`

> Can we use os and path in combination? Eg. create link using Path but read in its real path using os or do you need to do everything by one way?

Yes, you can use os and pathlib together, but it is nice to be consistent and pathlib is the more modern replacement. In particular, pathlib handles `/` vs `\` on *nix vs windows nicely so you don't have to think about the path separator. Everything you formerly did with os.path should be doable with pathlib I think... So I try to only use pathlib when I'm writing new code.

I am often looking at this reference for the corresponding uses: https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module

Reply:
I am trying those changes. Yeah, I saw that comparison link and that was helpful to look up how it would map between os and Path.